### PR TITLE
DR dashboard change for the imperative applications

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
Jira link: https://issues.redhat.com/browse/RHSTOR-4925

### CODE REVIEW

File: locales/en/plugin__odf-console.json [hunk 1]
```diff
@@ -140,15 +142,21 @@
...
+  "Protected volumes": "Protected volumes",
   "{{ pvcsWithIssueCount }} with issues": "{{ pvcsWithIssueCount }} with issues",
   "Operator status": "Operator status",
   "Cluster operator": "Cluster operator",
+  "in Disaster recovery relationship": "in Disaster recovery relationship",
   "{{ healthy }} healthy": "{{ healthy }} healthy",
   "{{ issues }} with issues": "{{ issues }} with issues",
-  "The applications count displays the total number of ApplicationSet and Subscription type applications in all disaster recovery configured clusters.": "The applications count displays the total number of ApplicationSet and Subscription type applications in all disaster recovery configured clusters.",
-  "{{ protected }} protected": "{{ protected }} protected",
+  "enrolled in Disaster recovery": "enrolled in Disaster recovery",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__odf-console.json:155]:
```suggestion
  "enrolled in disaster recovery": "enrolled in Disaster recovery",
```

**Comment by SanjalKatiyar** [comment]
> DB2
>
nits: icons are not aligned between `ACM discovered applications:` & `ACM managed applications:` rows... will look better if icons can be rendered towards flex-end so that "text" length won't affect it.

File: packages/mco/components/mco-dashboard/disaster-recovery/cluster-app-card/application.tsx [hunk 2]
```diff
@@ -133,13 +137,62 @@ export const ActivitySection: React.FC<CommonProps> = ({
...
+          lastKubeObjectSyncTime,
+          fromNow(lastKubeObjectSyncTime)
+        )
+      );
+    } else {
+      setLastSyncTime('N/A');
+    }
+  }, [lastKubeObjectSyncTime]);
+
+  useScheduler(updateSyncTime);
```
Comment (SanjalKatiyar) [inline packages/mco/components/mco-dashboard/disaster-recovery/cluster-app-card/application.tsx:175]: do we need separate timer for `Volume snapshot` and `KubeObject snapshot` ?
we can add a wrapper FC which will call `useScheduler(updateSyncTime)`, can will render/return both `VolumeSnapshotSection` & `KubeObjectSnapshotSection`.

Comment (SanjalKatiyar) [reply packages/mco/components/mco-dashboard/disaster-recovery/cluster-app-card/application.tsx:175]: similarly `SnapshotSection` FC will have a single `useState` storing an object.

File: packages/mco/components/mco-dashboard/disaster-recovery/cluster-app-card/cluster.tsx [hunk 3]
```diff
@@ -180,45 +244,46 @@ export const ApplicationsSection: React.FC<ApplicationsSectionProps> = ({
...
+        [0, 0]
+      ) || [0, 0],
     [clusterResources, clusterName, lastSyncTimeData]
   );
 
-  const totalAppSetsCount = clusterResources[clusterName]?.totalAppCount;
   const protectedAppCount =
     clusterResources[clusterName]?.protectedApps?.length;
+
+  // All discovred are protected apps
```
Comment (SanjalKatiyar) [inline packages/mco/components/mco-dashboard/disaster-recovery/cluster-app-card/cluster.tsx:268]:
```suggestion
  // All discovered are protected apps
```

File: packages/mco/components/mco-dashboard/disaster-recovery/cluster-app-card/common.tsx [hunk 4]
```diff
@@ -290,20 +325,60 @@ const AppDropdown: React.FC<Partial<ClusterAppDropdownProps>> = ({
...
+                appNS !== RAMEN_PROTECTED_APPS_NAMESPACE && (
+                  <SelectGroupNext
+                    key={appNS}
+                    label={t('Namespace: ') + `${appNS}`}
+                  >
+                    <SelectListNext>
+                      {options[appNS]?.map((appName: string) => (
+                        <SelectOptionNext
+                          key={appNS + appName}
+                          value={`${appNS}%#%${appName}`}
```
Comment (SanjalKatiyar) [inline packages/mco/components/mco-dashboard/disaster-recovery/cluster-app-card/common.tsx:366]: nits: better to store `%#%` in a variable on top of this file, used at many places...

File: packages/mco/components/mco-dashboard/disaster-recovery/summary-card/summary-card.tsx [hunk 5]
```diff
@@ -83,10 +91,13 @@ export const SummaryCard: React.FC = () => {
               <Text className="text-muted mco-dashboard__statusText--margin mco-dashboard__statusText--size">
                 {t('Clusters')}
               </Text>
-              <Text className="mco-dashboard__statusText--margin mco-summary__countText--size mco-dashboard__statusText--weight">
+              <Text className="mco-dashboard__statusText--margin pf-u-font-size-4xl mco-dashboard__statusText--weight">
```
Comment (SanjalKatiyar) [inline packages/mco/components/mco-dashboard/disaster-recovery/summary-card/summary-card.tsx:94]:
```suggestion
              <Text className="mco-dashboard__statusText--margin pf-v5-u-font-size-4xl mco-dashboard__statusText--weight">
```

File: packages/mco/components/mco-dashboard/disaster-recovery/summary-card/summary-card.tsx [hunk 6]
```diff
@@ -83,10 +91,13 @@ export const SummaryCard: React.FC = () => {
...
                 {summaryMap.clusters.totalCount}
               </Text>
-              <Divider className="mco-summary__divider--width mco-dashboard__statusText--margin" />
+              <Text className="text-muted mco-dashboard__statusText--margin">
+                {t('in Disaster recovery relationship')}
+              </Text>
+              <Divider className="pf-u-w-75 mco-dashboard__statusText--margin" />
```
Comment (SanjalKatiyar) [inline packages/mco/components/mco-dashboard/disaster-recovery/summary-card/summary-card.tsx:100]:
```suggestion
              <Divider className="pf-v5-u-w-75 mco-dashboard__statusText--margin" />
```

File: packages/mco/components/mco-dashboard/disaster-recovery/summary-card/summary-card.tsx [hunk 7]
```diff
@@ -107,21 +118,22 @@ export const SummaryCard: React.FC = () => {
...
-                  <Trans t={t}>
-                    The applications count displays the total number of
-                    ApplicationSet and Subscription type applications in all
-                    disaster recovery configured clusters.
-                  </Trans>
-                </FieldLevelHelp>
               </Text>
-              <Text className="mco-dashboard__statusText--margin mco-summary__countText--size mco-dashboard__statusText--weight">
-                {summaryMap.applications.totalCount}
+              <Text className="mco-dashboard__statusText--margin pf-u-font-size-4xl mco-dashboard__statusText--weight">
```
Comment (SanjalKatiyar) [inline packages/mco/components/mco-dashboard/disaster-recovery/summary-card/summary-card.tsx:122]:
```suggestion
              <Text className="mco-dashboard__statusText--margin pf-v5-u-font-size-4xl mco-dashboard__statusText--weight">
```

File: locales/en/plugin__odf-console.json (hunk 1 above)
Comment (SanjalKatiyar) [reply locales/en/plugin__odf-console.json:155]: same for `in disaster recovery relationship` IMO...

File: locales/en/plugin__odf-console.json [hunk 8]
```diff
@@ -124,9 +126,9 @@
   "Operators health": "Operators health",
   "Peer connection": "Peer connection",
   " {{ peerConnectedCount }} Connected": " {{ peerConnectedCount }} Connected",
-  "Total applications": "Total applications",
-  " {{ protectedAppCount }} protected": " {{ protectedAppCount }} protected",
-  "{{ appsWithIssues }} of {{ protectedAppCount }} with issues": "{{ appsWithIssues }} of {{ protectedAppCount }} with issues",
+  "Protected Applications": "Protected Applications",
```
[the remark this answers is not in the document]
Comment (GowthamShanmugam) [reply locales/en/plugin__odf-console.json:129]: ack

File: locales/en/plugin__odf-console.json (hunk 1 above)
Comment (GowthamShanmugam) [reply locales/en/plugin__odf-console.json:155]: ack

**Comment by GowthamShanmugam** [comment]
image

**Comment by SanjalKatiyar** [comment]
need to add unit test for `disaster-recovery/parsers/discovered-parser.tsx` cc @GowthamShanmugam ... we can do so in a follow-up PR... tagging this one for now...

**Pull request merged**